### PR TITLE
Add new get finance page

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/TT-348)

# feature flag on

image

# feature flag off

image

### CODE REVIEW

File: finance/views.py [hunk 1]
```diff
@@ -49,6 +64,14 @@ def get_context_data(self, *args, **kwargs):
 
 
 class GetFinanceStartRedirectView(FeatureFlagMixin, RedirectView):
+    """
+    Using a redirect URL here instead of putting a hyperlink directly to Pardot
+    in the html. the reason is go Google links to this page instead of Pardot.
```
Comment (konradko) [inline finance/views.py]: whut

File: finance/views.py [hunk 2]
```diff
@@ -49,6 +64,14 @@ def get_context_data(self, *args, **kwargs):
...
+    The reason that is desirable is in future we may change how we track, or
```
Comment (konradko) [inline finance/views.py]: whut

File: finance/views.py [hunk 3]
```diff
@@ -27,13 +27,28 @@ def get_context_data(self, *args, **kwargs):
...
     template_name = 'finance/get_finance.html'
     slug = EXPORT_READINESS_GET_FINANCE_SLUG
 
 
+class GetFinanceNegotiator(TemplateView):
+    def __new__(cls, *args, **kwargs):
+        if settings.FEATURE_FLAGS['UKEF_LEAD_GENERATION_ON']:
+            return GetFinance(*args, **kwargs)
+        else:
+            return DeprecatedGetFinance(*args, **kwargs)
```
Comment (konradko) [inline finance/views.py]: Not a fan of the name, we should have deprecated code around, but this is actually still in use depending on the flag... Not a big deal.

File: finance/views.py (hunk 1 above)
Comment (richtier) [reply finance/views.py]: as per our convo I turned a github comment into a code comment. will edit the comment to make ti clearer.

File: finance/views.py (hunk 3 above)
Comment (richtier) [reply finance/views.py]: marking code as deprecated is a step away from deleting it e.g, the deprecation warnings in Django when using the old style middleware.

`DeprecatedGetFinance` is used when the feature flag is turned off, and `DeprecatedGetFinance` will be deleted once the feature is to be turned on always.

**Pull request merged**